src/components/MapView/Analyser/index.tsx

```diff
@@ -107,17 +134,22 @@ function Analyser({ extent, classes }: AnalyserProps) {
 
   // set default date after dates finish loading and when hazard layer changes
   useEffect(() => {
-    const dates = hazardLayerId
-      ? availableDates[
-          (LayerDefinitions[hazardLayerId] as WMSLayerProps)?.serverLayerName
-        ]
-      : null;
-    if (!dates || dates.length === 0) {
+    if (
+      availableHazardDates === undefined ||
+      availableHazardDates.length === 0
+    ) {
       setSelectedDate(null);
+      setStartDate(null);
+      setEndDate(null);
     } else {
-      setSelectedDate(dates[dates.length - 1]);
+      const lastDate = availableHazardDates[
+        availableHazardDates.length - 1
+      ].getTime();
+      setSelectedDate(lastDate);
+      setStartDate(lastDate);
+      setEndDate(lastDate);
     }
-  }, [availableDates, hazardLayerId]);
+  }, [availableDates, availableHazardDates?.toString(), hazardLayerId]);
```

Comment:
It seems very odd to have `?toString()` in the dependency, what are you trying to achieve?

Reply:
I used it because availableHazardDates is being created on each render, deriving from availableDates and hazardLayerId (which are both stored in state).  When I passed the array object of availableHazardDates into a useEffect dependency array, it automatically triggers a new call to the useEffect callback because React thinks the dependencies changed.  Although the values in availableHazardDates are in effect the same, React thinks they are different because useEffect is using a "shallow comparison".  I evaluated a few different options.

1) store availableHazardDates in a useState.  This isn't a bad solution, but each call to useState triggers a new render cycle, so I figured it was best to save one render cycle by deriving it from the state of availableDates and hazardLayerid.
2) implement [useDeepCompareEffect](https://github.com/kentcdodds/use-deep-compare-effect).  I've used this library successfully before and the maintainer is reliable, but I felt it wasn't necessary to add a dependency just for a small purpose.  It might be worth considering in the future though.
3) I could have also used JSON.stringify instead of toString.  JSON.stringify is traditionally thought of as kinda slow, so I figured toString is faster.  toString sometimes stringify objects poorly, but I checked it and it seems to work for dates well.  It just adds a comma between the dates.

Happy to discuss alternatives, too.

Reply:
see comment above, basically calculated lastDate outside useEffect, so could pass shallow number to useEffect instead of array of dates.